src/Coin.cpp

```diff
@@ -55,10 +55,12 @@ std::string TW::loadAddress(TWCoinType coin, const Data& data) {
     case TWCoinTypeThunderToken:
     case TWCoinTypeTomoChain:
     case TWCoinTypeVeChain:
-    case TWCoinTypeWanChain:
     case TWCoinTypeXDai:
         return Ethereum::Address(data).string();
 
+    case TWCoinTypeWanChain:
+        return Ethereum::Address(data, TWEthereumChecksumTypeWanchain).string();
```

Comment:
+ you need to add validate address handler too. either all lowercase or checksummed address

Reply:
`Ethereum::isValid` checks only size `data.size() == size`

Reply:
We should have a `WanchainAddress` instead of adding a flag to EthereumAddress